tools/alphafill_cofactor_survey/find_cofactor.py

```diff
@@ -69,22 +72,116 @@ def load_filter_file(filepath):
     with open(filepath) as f:
         return set(line.strip().upper() for line in f if line.strip())
 
+def calculate_ligand_contacts(ligand, protein_residues):
+    """ Calculate minimum distance and contacts for a ligand """
+    min_distance = 100.0
+    contacts = []
+
+    for ligand_atom in ligand:
+        for protein_res in protein_residues:
+            for protein_atom in protein_res:
+                distance = np.linalg.norm(ligand_atom.coord - protein_atom.coord)
+                if distance < 4.0:
+                    contact_str = f"{protein_res.resname}{protein_res.id[1]}"
+                    if contact_str not in contacts:
+                        contacts.append(contact_str)
+                if distance < min_distance:
+                    min_distance = distance
+    return min_distance, contacts
+
 def main():
+
     parser = argparse.ArgumentParser(description="Query AlphaFill for cofactors, with optional filtering.")
     parser.add_argument("-u", help="Single UniProt ID")
     parser.add_argument("-i", help="Index file with dataset, should have column 'UniProt AC'")
     parser.add_argument("-f", "--filter", help="Optional file with valid cofactor IDs")
     parser.add_argument("-o", "--output", default="summary_output", help="Output file prefix")
+    parser.add_argument("-p", "--pdb", help="PDB file to analyze for ligands")
+    parser.add_argument("-s", "--start", type=int, help="Start residue of trimmed region")
+    parser.add_argument("-e", "--end", type=int, help="End residue of trimmed region")
+    parser.add_argument("-c", "--cofactors", help="JSON file with cofactor definitions")
```

Comment:
I think we should also add an input flag and mechanism to select the protein chain we are interested in; otherwise the start and end flags work for every chain (which is not what we want)